sdk/core/azure-core/azure/core/tracing/opentelemetry.py

```diff
@@ -0,0 +1,219 @@
+        schema_url: Optional[str] = None,
+        attributes: Optional[Attributes] = None,
+    ) -> None:
+        self._tracer = trace.get_tracer(
+            instrumenting_module_name=library_name or _DEFAULT_MODULE_NAME,
+            instrumenting_library_version=library_version or VERSION,
+            schema_url=schema_url or _DEFAULT_SCHEMA_URL,
+            attributes=attributes,
+        )
+
+    def start_span(
+        self,
+        name: str,
+        *,
+        kind: _SpanKind = _SpanKind.INTERNAL,
+        attributes: Optional[Attributes] = None,
+        links: Optional[Sequence[_Link]] = None,
+    ) -> Span:
+        """Starts a span without setting it as the current span in the context.
+
+        :param name: The name of the span
+        :type name: str
+        :keyword kind: The kind of the span. INTERNAL by default.
+        :paramtype kind: ~azure.core.tracing.SpanKind
+        :keyword attributes: Attributes to add to the span.
+        :paramtype attributes: Mapping[str, AttributeValue]
+        :keyword links: Links to add to the span.
+        :paramtype links: list[~azure.core.tracing.Link]
+        :return: The span that was started
+        :rtype: ~azure.core.tracing.Span
+        """
+        otel_kind = _KIND_MAPPINGS.get(kind, OpenTelemetrySpanKind.INTERNAL)
+        otel_links = self._parse_links(links)
+
+        otel_span = self._tracer.start_span(
+            name,
+            kind=otel_kind,
+            attributes=attributes,
+            links=otel_links,
+            record_exception=False,
+        )
+
+        return otel_span
+
+    @contextmanager
+    def start_as_current_span(
+        self,
+        name: str,
+        *,
+        kind: _SpanKind = _SpanKind.INTERNAL,
+        attributes: Optional[Attributes] = None,
+        links: Optional[Sequence[_Link]] = None,
+    ) -> Iterator[Span]:
+        """Context manager that starts a span and sets it as the current span in the context.
+
+        Exiting the context manager will call the span's end method.
+
+        .. code:: python
+
+            with tracer.start_as_current_span("span_name") as span:
+                # Do something with the span
+                span.set_attribute("key", "value")
+
+        :param name: The name of the span
+        :type name: str
+        :keyword kind: The kind of the span. INTERNAL by default.
+        :paramtype kind: ~azure.core.tracing.SpanKind
+        :keyword attributes: Attributes to add to the span.
+        :paramtype attributes: Optional[Attributes]
+        :keyword links: Links to add to the span.
+        :paramtype links: Optional[Sequence[Link]]
+        :return: The span that was started
+        :rtype: ~opentelemetry.trace.Span
+        """
+        span = self.start_span(name, kind=kind, attributes=attributes, links=links)
+        with trace.use_span(span, record_exception=False, end_on_exit=True) as span:  # type: ignore[attr-defined]  # pylint: disable=not-context-manager
```

Comment:
just curious, for `record_exception` and `end_on_exit`, is there a reason we decided to pass their values instead of making them kwonly on the method?

Reply:
Yes. Mainly asking because there was a time that I wanted to control the end_on_exit option. But it's additive to add more knobs so I think we can wait for feedback.

Reply:
do other langs have it and are there cases where people use this ?

Reply:
@kashifkhan for other languages (at least .NET and Java), they are not recording exceptions as span events. There is still some exception info in the span status description, and error.type attribute (some more context here: https://github.com/Azure/azure-sdk-for-python/pull/39563#discussion_r1974501508)). For  `end_on_exit`, since this is context manager specific, it's python-only. As for cases where people use `end_on_exit`, I haven't seen a need for it for the general SDK tracing needs, but seems like there could be a desire for it in SDKs doing their own custom tracing. For this, I'd like to get more data, and do a follow-up PR.